### ISSUE TITLE
Optional checkbox: UX is not optimal

### ISSUE THREAD

**Issue by nitriques** [symphonycms/symphony-2#2759]
Affected Symphony version(s) : 2.6.11, 2.7.0, 3.0.0

I've always found it weird to have the Optional labels on checkbox fields.

I mean, the checkbox is either on, off or null. We interpret null like off. It is not "optional".

I would propose to switch this label the other way around, i.e., show a label on "required" fields. I also think that all fields should to the same.

I would propose to

1. Remove the optional label for 2.7.1
1. Switch to "required" labels in 3.0.0

**Comment by nilshoerrmann** [comment]
Just to let you know: This has been discussed years ago, too, and there is no such thing as "required" labels in Symphony (there is only _optional_: anything that's not optional is _required_). "required" is only a term within the section editor. So it would be strange to have some fields stating they are optional and some that are required while those without labels are required by default.

**Comment by nilshoerrmann** [comment]
Ah, and I just saw the attached milestone: from an UI/UX this would be a breaking change requiring a major release (currently `3.0.0`).

**Comment by nitriques** [comment]
> So it would be strange to have some fields stating they are optional and some that are required while those without labels are required by default.

Yeah I know, But even I have problems with the concept of an optional checkbox. Symphony is the only place I've seen this.

I've target for 2.7.1, but only if we can find something that is not a breaking change. Like maybe just remove the label altogether. I never even used a "required" checkbox.

**Comment by nilshoerrmann** [comment]
I totally understand. Nevertheless, it should not be fixed until Symphony 3.0.0 in my eyes. Otherwise, you'd make a confusing interface decision even more confusing due to the resulting incoherence.

**Comment by nilshoerrmann** [comment]
Maybe it's a good idea to finally separate core (PHP API) and UI so both could be versioned separatelly.

**Comment by michael-e** [comment]
> I never even used a "required" checkbox.

Oh, I have used this!

* [ ] I accept the terms and conditions. I know this will be expensive.

:-)

**Comment by michael-e** [comment]
I agree with @nilshoerrmann on this topic. We shouldn't apply such changes to 2.7.

**Comment by nitriques** [comment]
> Maybe it's a good idea to finally separate core (PHP API) and UI so both could be versioned separatelly.

It's in the long term plans...

> Oh, I have used this!

I knew you would have ;) It is a valid feature, but its maximum 1% of the uses.

> I agree with @nilshoerrmann on this topic. We shouldn't apply such changes to 2.7.

Ok then.
